src/lib.rs

```diff
@@ -0,0 +1,1167 @@
+                        self.range.start = bs + 1;
+                        return Some(bs);
+                    }
+                }
+                i = b * bits_per_block::<T>();
+            }
+        }
+        self.range.start = self.range.end;
+        None
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.range.size_hint()
+    }
+}
+
+#[derive(Clone)]
+pub struct IterUnsetBits<'a, T: 'a> {
+    vob: &'a Vob<T>,
+    range: Range<usize>,
+}
+
+impl<'a, T: Debug + One + PrimInt + Zero> Iterator for IterUnsetBits<'a, T> {
+    type Item = usize;
+
+    fn next(&mut self) -> Option<usize> {
+        if let Some(mut i) = self.range.next() {
+            for b in storage_block_off::<T>(i) .. storage_size_align::<T>(self.vob.len) {
+                let v = self.vob.vec[b];
+                if v != T::max_value() {
+                    let i_off = i % bits_per_block::<T>();
+                    let tz = (!v >> i_off).trailing_zeros() as usize;
+                    if tz < bits_per_block::<T>() {
+                        let bs = b * bits_per_block::<T>() + i_off + tz;
+                        self.range.start = bs + 1;
+                        if bs >= self.vob.len {
+                            break;
+                        }
+                        return Some(bs);
+                    }
+                }
+                i = b * bits_per_block::<T>();
+            }
+        }
+        self.range.start = self.range.end;
+        None
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.range.size_hint()
+    }
+}
+
+impl<T: Debug + One + PrimInt + Zero> PartialEq for Vob<T> {
+    fn eq(&self, other: &Self) -> bool {
+        if self.len != other.len {
+            return false;
+        }
+        self.iter_storage()
+            .zip(other.iter_storage())
+            .all(|(w1, w2)| w1 == w2)
+    }
+}
+
+impl<T: Debug + One + PrimInt + Zero> Eq for Vob<T> {}
+
+impl<T :Debug + Hash + One + PrimInt + Zero> Hash for Vob<T> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.len.hash(state);
+        for b in self.iter_storage() {
+            b.hash(state);
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct StorageIter<'a, B: 'a> {
+    iter: slice::Iter<'a, B>,
+}
+
+impl<'a, T: Debug + One + PrimInt + Zero> Iterator for StorageIter<'a, T> {
+    type Item = T;
+
+    fn next(&mut self) -> Option<T> {
+        self.iter.next().cloned()
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.iter.size_hint()
+    }
+}
+
+/// How many bits are stored in each underlying storage block?
+fn bits_per_block<T>() -> usize {
+    size_of::<T>() * 8
+}
+
+/// Return the offset in the vector of the storage block storing the bit `off`.
+fn storage_block_off<T>(off: usize) -> usize {
+    off / bits_per_block::<T>()
+}
+
+/// Takes as input a number of bits requiring storage; returns an aligned number of blocks needed
```

Comment:
OK, I understand a comment you made on Jake's repo now.

I'd call this `pad_to_blocksize` or something, as it's clearer what the intention is.I thought you were optimising by ensuring the storage sits on an alignment boundary.

And, yes, I know wikipedia lists this use case as one of the three kinds of alignment :P

Reply:
`pad` to me suggests "actively change the size"... :/

Reply:
Hrm, yeah?

OK, how about `storage_bytesize`.

Reply:
That's not right. One moment.

Reply:
`num_blocks_required`, perfect!

Reply:
`storage_blocks_required`?

Reply:
sure!

Reply:
OK, commit with that change in pushed.